free-2.md

```diff
@@ -58,12 +105,25 @@ making language design decisions), as opposed to "systems" aspects of the
 project (e.g., implementing a complicated semantics that doesn't require a lot
 of language design)?_
 
+A lot of the time would be spent on figuring out good ways to interact with the
+user. Keeping track of events is pretty easy and notifications can be anywhere
+from easy to pretty difficult. However, the user interaction can also be pretty
+incremental: adding support for manual input would be one level, gmail input
+would be harder, and other methods of input and output could get trickier.
 
 ### Scope
 _How big an idea is this? How ambitious is this project?_
 
+This is potentially a big project depending on how many different devices and
+platforms it supports, but could pretty easily be toned down to something
+reasonable. If it does get big and is done well, it could be very useful and I
+could definitely see myself using it.
 
 ### Benefits and drawbacks
 _Why might this be a good idea for a project? Why might this not be a good idea 
 project?_
 
+This is a good idea for a project because it is very language-y and useful.
+There are some drawbacks in that supporting enough different methods of input
+and output could be technically difficult, but that is all part of the language
+design.
```

Comment:
A language (e.g., a composable data structure + algorithms over that structure) may be an appropriate _encoding_ of this domain. I'm not sure that text-based language is the _best_ interface for this domain. I could be wrong (and I'd very much like to be proved so), but I think a text-based interface would be quickly overwhelmed by the complexity of the domain.

The interface(s) you propose may indeed be right on. Alas, they're not linguistic interfaces. If you went ahead with this project, I'd want to make sure that most of your time was on language design, rather than parsing emails, etc. Alternatively, if you wanted to design a syntax that was sort of like natural language--perhaps a stylized agenda that could be sent in an email--that could be interesting and challenging.

It's not yet clear what happens when a program runs, specifically. If you choose this project, it would be good to do some early research on algorithms to compute what you propose.